File: dart/lib/src/env_config.dart
import 'dart:io';

/// อ่าน .env ทุกครั้งที่ถูกเรียก — live reload เหมือน Rust dotenvy per-request
String? readFromDotenv(String key) {
  try {
    final lines = File('.env').readAsLinesSync();
    for (final line in lines) {
      final trimmed = line.trim();
      if (trimmed.isEmpty || trimmed.startsWith('#')) continue;
      final eqIdx = trimmed.indexOf('=');
      if (eqIdx == -1) continue;
      final k = trimmed.substring(0, eqIdx).trim();
      final v = trimmed.substring(eqIdx + 1).trim();
      if (k == key) return v;
    }
  } catch (_) {}
  return null;
}

/// priority: .env file > Platform.environment > fallback (mirrors Rust resolve_config)
String resolveConfig(String key, String fallback) {
  return readFromDotenv(key) ?? Platform.environment[key] ?? fallback;
}

File: csharp/src/EnvConfig.cs
namespace CSharpService;

public static class EnvConfig
{
    public static string? ReadFromDotenv(string key)
    {
        if (!File.Exists(".env")) return null;
        foreach (var line in File.ReadAllLines(".env"))
        {
            var trimmed = line.Trim();
            if (trimmed.StartsWith('#') || !trimmed.Contains('=')) continue;
            var idx = trimmed.IndexOf('=');
            var k = trimmed[..idx].Trim();
            var v = trimmed[(idx + 1)..].Trim();
            if (k == key) return v;
        }
        return null;
    }

    public static string ResolveConfig(string key, string fallback)
    {
        return ReadFromDotenv(key)
            ?? Environment.GetEnvironmentVariable(key)
            ?? fallback;
    }

    public static int Add(int a, int b) => a + b;
}
